sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -681,11 +681,14 @@ class SparkSqlAstBuilder extends AstBuilder {
     } else {
       val serdeInfo = maybeSerdeInfo.get
       if (serdeInfo.storedAs.isEmpty) {
+        val serde = serdeInfo.formatClasses
+          .flatMap(f => HiveSerDe.getSerDeFromFormatClasses(f.input, f.output))
+          .orElse(serdeInfo.serde)
```

Comment:
What are the expected behaviors of both FormatClasses and ROW FORMAT SERDE are given but conflict

Reply:
Thanks for your review. I have made some adjustments to prioritize the user-defined 'ROW FORMAT SERDE'.